### PR TITLE
Adds LOVES_BOOKS and HATES_BOOKS traits

### PR DESCRIPTION
**Pull request by Inglonias**
After some discussion on Discord about audio books and how they might make Illiterate a little less unbearable, I decided instead to make this pair of traits. It's not actually related to Illiterate, but hey, the discussion helped me think of the traits.

**Two new traits: "Bookworm" and "Hates Books"**

**Bookworm:** Adds a flat +1 bonus to all books you read.

**Hates Books:** Reduces fun from reading books by one and then caps fun from books at zero, making reading fiction for fun a complete waste of time.

This does not interact with Spiritual or Cannibal bonuses when reading, regardless of what your inventory says when you examine the books. The game already doesn't check those traits when determining how much fun a book will be, even though I suppose that it can now with the new method I added. I guess that's another PR.

### CODE REVIEW

**Comment by MT-Arnoldussen** [comment]
Both traits should conflict with ILLITERATE and vice versa

**Comment by MT-Arnoldussen** [comment]
Also I'm not sure if HATES_BOOKS is really worth 2 points. It doesn't really affect skills books at all, unlike LOVES_BOOKS.

**Comment by Inglonias** [comment]
Alright. I've added the conflict with Illiterate, and you're right. HATES_BOOKS isn't a big enough drawback to give you two points. If it reduced the fun of all books AND capped it at zero, that might be something. I'm thinking of doing that.

**Comment by Xhuis** [comment]
Bookworm also seems pretty pricey for its benefit. I'd personally weight it at a cost of 1 point, along with traits like Psychopath, since its benefit isn't super useful.

Speaking of 1-point traits, how does this interact with Spiritual? Do you become the `ultimate religion man` and get +100 morale from reading a Bible?

**Comment by Inglonias** [comment]
This doesnt interact with Spiritual or Cannibal, which are the pther traits
that modify the fun you get out of a book. Those traits override mine.

- Joshua Nyden -

File: src/game_inventory.cpp [hunk 1]
```diff
@@ -641,11 +641,11 @@ class read_inventory_preset: public pickup_inventory_preset
                 return unlearned > 0 ? to_string( unlearned ) : std::string();
             }, _( "RECIPES" ), unknown );
 
-            append_cell( [ this ]( const item_location & loc ) -> std::string {
+            append_cell( [ p , this ]( const item_location & loc ) -> std::string {
```
Comment (DracoGriffin) [inline src/game_inventory.cpp]: `p , this` may have an extra space. I believe it's supposed to be `p, this`

**Comment by FulcrumA** [comment]
I am unsure about just morale boost from loving books. Usually what matters is content of the book. Reading bothersome word salad won't be more enjoyable just because it's in a book format. Perhaps rather than just morale effects, some modifier to focus would be good? If someone's hating reading, they'll probably have harder time concentrating on the activity and quite the opposite for avid reader used to such activity.

**Comment by Inglonias** [comment]
Morale directly affects focus, so that's sort of what I was going for there.

**Comment by Inglonias** [comment]
I feel that this PR is now ready to merge. To sum up, here's what it does:

* Adds a trait you can take as an advantage: "Bookworm", which costs 1 point and flatly increases the joy of books by one.
* Adds a trait you can take as a disadvantage: "Hates books", which earns you 1 point and reduces the joy of books by one, then setting it to zero if it's still positive.
* It does this by creating the player method book_fun_for, which is similar to the comestible method "fun_for" that already existed to accommodate the Gourmand trait. All fun checks for books now feed through this new method, allowing us to do a few minor things like display the true fun of "To Serve Man" for cannibals, or the true fun for religious texts for spiritual characters on the inventory screen. This PR doesn't do that, but that's my next task if this is merged.

File: data/json/mutations.json [hunk 2]
```diff
@@ -292,7 +292,18 @@
...
         "valid" : false,
         "active" : true
-    },{
+    },
+    {
+        "type" : "mutation",
+        "id" : "LOVES_BOOKS",
+        "name" : "Bookworm",
+        "points" : 1,
+        "description" : "There's nothing quite like the smell of a good book! Books are more fun (or less boring) for you!",
```
Comment (ZhilkinSerg) [inline data/json/mutations.json]: Two whitespaces between sentences.

File: data/json/mutations.json [hunk 3]
```diff
@@ -692,7 +703,18 @@
...
         "valid" : false,
         "cancels" : ["VEGETARIAN"]
-    },{
+    },
+    {
+        "type" : "mutation",
+        "id" : "HATES_BOOKS",
+        "name" : "Hates Books",
+        "points" : -1,
+        "description" : "Reading is for nerds! Boring books are more boring, and you can't have fun by reading books.",
```
Comment (ZhilkinSerg) [inline data/json/mutations.json]: Two whitespaces between sentences.

File: src/game_inventory.cpp [hunk 4]
```diff
@@ -641,11 +641,11 @@ class read_inventory_preset: public pickup_inventory_preset
                 return unlearned > 0 ? to_string( unlearned ) : std::string();
             }, _( "RECIPES" ), unknown );
 
-            append_cell( [ this ]( const item_location & loc ) -> std::string {
+            append_cell( [ p, this ]( const item_location & loc ) -> std::string {
```
Comment (codemime) [inline src/game_inventory.cpp]: This needlessly copies `p` (`player` is expensive to copy). Should be `&p`.

Comment (Inglonias) [reply src/game_inventory.cpp]: I'll fix this as soon as I can. Thanks for telling me.

**Pull request merged**